File: StudyGamePrograming06/Sphere.cpp
#include "Sphere.h"
#include "MeshComponent.h"
#include "Game.h"
#include "Renderer.h"
#include "MoveComponent.h"

Sphere::Sphere(Game* game) :Actor(game)
{
	SetPosition(Vector3(-500.0f, 200.0f, 0.0f));
	Quaternion q(Vector3::UnitX, Math::ToRadians(-23.5f));
	SetRotation(q);
	SetScale(5.0f);
	MeshComponent* mc = new MeshComponent(this);
	mc->SetMesh(game->GetRenderer()->GetMesh("Assets/Sphere.gpmesh"));
	mMoveComp = new MoveComponent(this);
}

void Sphere::UpdateActor(float deltaTime)
{
	Actor::UpdateActor(deltaTime);

	Vector3 axis = Vector3(0.0f, Math::Cos(Math::ToRadians(66.5f)), Math::Sin(Math::ToRadians(66.5f)));
	float rotSpeedMax = 5.0f * Math::Pi;
	float rotSpeed = rotSpeedMax * deltaTime;
	mMoveComp->SetRotSpeed(rotSpeed * axis);
}

Sphere2::Sphere2(Game* game) :Actor(game)
{
	SetPosition(Vector3(-500.0f, -200.0f, 0.0f));
	Quaternion q(Vector3::UnitX, Math::ToRadians(-23.5f));
	SetRotation(q);
	SetScale(5.0f);
	MeshComponent* mc = new MeshComponent(this);
	mc->SetMesh(game->GetRenderer()->GetMesh("Assets/Sphere2.gpmesh"));
	mMoveComp = new MoveComponent(this);
}

void Sphere2::UpdateActor(float deltaTime)
{
	Actor::UpdateActor(deltaTime);

	Vector3 axis = Vector3(0.0f, Math::Cos(Math::ToRadians(66.5f)), Math::Sin(Math::ToRadians(66.5f)));
	float rotSpeedMax = 5.0f * Math::Pi;
	float rotSpeed = rotSpeedMax * deltaTime;
	mMoveComp->SetRotSpeed(rotSpeed * axis);
}

File: StudyGamePrograming06_VB/Sphere.vb
Imports OpenTK.Mathematics
Imports OpenTK.Windowing.GraphicsLibraryFramework
Public Class Sphere
    Inherits Actor
    'public
    Sub New(ByRef game As Game)
        MyBase.New(game)
        SetPosition(New Vector3(200.0, 100.0, 0.0))
        Dim q As Quaternion = New Quaternion(Vector3.UnitX, -23.5 / 180.0 * Math.PI)
        SetRotation(q)
        SetScale(5.0)
        Dim mc As MeshComponent = New MeshComponent(Me)
        mc.SetMesh(game.GetRenderer().GetMesh("Assets/Sphere.gpmesh"))
        mMoveComp = New MoveComponent(Me, 20)
    End Sub

    Public Overrides Sub UpdateActor(deltaTime As Single)
        MyBase.UpdateActor(deltaTime)
        Dim mAxis As Vector3 = New Vector3(0.0, Math.Cos(66.5 / 180.0 * Math.PI), Math.Sin(66.5 / 180.0 * Math.PI))
        Dim rotSpeedMax As Single = 5.0 * Math.PI
        Dim rotSpeed As Single = rotSpeedMax * deltaTime
        mMoveComp.SetRotSpeed(rotSpeed * mAxis)
    End Sub

    'private
    Dim mMoveComp As MoveComponent
End Class
